### PR TITLE
curl: add libssh2 in PACKAGECONFIG list to enable sftp

### PR DESCRIPTION
**Pull request by texierp**
This is useful if we want to use an sftp server with SWUpdate

Signed-off-by: Pierre-Jean Texier <pjtexier@koncepto.io>

### CODE REVIEW

File: recipes-core/curl/curl_%.bbappend [hunk 1]
```diff
@@ -0,0 +1,3 @@
+FILESEXTRAPATHS_prepend := "${THISDIR}/${PN}:"
```
Comment (jorisoffouga) [inline recipes-core/curl/curl_%.bbappend]: I think FILESEXTRAPATHS_prepend is not necessary, you don't add any file

**Review by jorisoffouga — COMMENTED** [review]
it would be better to use PACKAGECONFIG_append_pn-curl = " libssh2" in site.conf to avoid adding a bbappend

File: recipes-core/curl/curl_%.bbappend (hunk 1 above)
Comment (texierp) [reply recipes-core/curl/curl_%.bbappend]: Indeed, fixed

**Comment by texierp** [comment]
The advantage of the bbappend file is that we are able to use the 'show-appends' command ;)

But also for consistency with https://github.com/bdx-iot/meta-swupdate-dev/blob/master/dynamic-layers/qt5-layer/recipes-qt/qt5/qtbase_%25.bbappend#L2

**Comment by jorisoffouga** [comment]
Indeed ;)

**Review by jorisoffouga — APPROVED** [review]

**Pull request merged**